Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+TypeSignatureDisambiguation.swift

```diff
@@ -0,0 +1,437 @@
+        // For `Set<Int>`, use the Swift Algorithms implementation to generate the possible combinations.
+        self.combinations(ofCount: 1...)
+    }
+}
+extension _TinySmallValueIntSet: _IntSet, _Combination {
+    // Explicitly specify the associated types for Swift 5.9 compatibility.
+    typealias Combination = Self
+    typealias CombinationSequence = [Combination]
+
+    func combinationsToCheck() -> [Self] {
+        // For `_TinySmallValueIntSet`, leverage the fact that bits of an Int represent the possible combinations.
+        let smallest = storage.trailingZeroBitCount
+        
+        var combinations: [Self] = []
+        combinations.reserveCapacity((1 << count /*known to be <64 */) - 1)
+        
+        for raw in 1 ... storage >> smallest {
+            let combination = Self(storage: UInt64(raw << smallest))
+            
+            // Filter out any combinations that include columns that are the same for all overloads
+            guard self.isSuperset(of: combination) else { continue }
+
+            combinations.append(combination)
+        }
+        // The bits of larger and larger Int values won't be in order of number of bits set, so we sort them.
+        return combinations.sorted(by: { $0.count < $1.count })
+    }
+}
+
+/// A specialized set-algebra type that only stores the possible values `0 ..< 64`.
+///
+/// This specialized implementation is _not_ suitable as a general purpose set-algebra type.
+/// However, because the code in this file only works with consecutive sequences of very small integers (most likely `0 ..< 16` and increasingly less likely the higher the number),
+/// and because the the sets of those integers is frequently accessed in loops, a specialized implementation addresses bottlenecks in `_minimalSuggestedDisambiguation(...)`.
+///
+/// > Important:
+/// > This type is thought of as file private but it made internal so that it can be tested.
+struct _TinySmallValueIntSet: SetAlgebra {
+    typealias Element = Int
+    
+    init() {}
+    
+    @usableFromInline
+    private(set) var storage: UInt64 = 0
+    
+    @inlinable
+    init(storage: UInt64) {
+        self.storage = storage
+    }
+    
+    private static func mask(_ number: Int) -> UInt64 {
+        precondition(number < 64, "Number \(number) is out of bounds (0..<64)")
+        return 1 << number
+    }
+    
+    @inlinable
+    @discardableResult
+    mutating func insert(_ member: Int) -> (inserted: Bool, memberAfterInsert: Int) {
+        let newStorage = storage | Self.mask(member)
+        defer {
+            storage = newStorage
+        }
+        return (newStorage != storage, member)
+    }
+    
+    @inlinable
+    @discardableResult
+    mutating func remove(_ member: Int) -> Int? {
+        let newStorage = storage & ~Self.mask(member)
+        defer {
+            storage = newStorage
+        }
+        return newStorage != storage ? member : nil
+    }
+    
+    @inlinable
+    @discardableResult
```

Comment:
Why do we need this behavior for returning the removed value? Just to conform to `SetAlgebra`? I don't see any explicit dependency on this here in this PR...?